### ISSUE TITLE
TypeError: Failed to parse URL from /api/logout

### ISSUE THREAD

**Issue by MendyLanda** [awinogrodzki/next-firebase-auth-edge#128]
I'm getting a type error when running build on a nextjs app (not interrupting but still annoying):
<img width="379" alt="image" src="https://github.com/awinogrodzki/next-firebase-auth-edge/assets/54242706/902fb40f-e814-45b9-8d74-03330a51d53b">

```bash
TypeError: Failed to parse URL from /api/logout
    at Object.fetch (node:internal/deps/undici/undici:11457:11)
    at async /Users/mendy/dev/@kangaroo/apps/school-web/.next/server/pages/auth/logout.js:1:3644 {
  [cause]: TypeError [ERR_INVALID_URL]: Invalid URL
      at new NodeError (node:internal/errors:399:5)
      at new URL (node:internal/url:560:13)
      at new Request (node:internal/deps/undici/undici:7041:25)
      at fetch2 (node:internal/deps/undici/undici:10598:25)
      at Object.fetch (node:internal/deps/undici/undici:11455:18)
      at fetch (node:internal/process/pre_execution:230:25)
      at /Users/mendy/dev/@kangaroo/apps/school-web/.next/server/pages/auth/logout.js:1:3650 {
    input: '/api/logout',
    code: 'ERR_INVALID_URL'
  }
}
```

Is this expected behaviour or am I doing something wrong? 

In the logout page I have this hook running:
```ts
import { useRouter } from "next/router";
import { signOut } from "@firebase/auth";

import { useFirebaseAuth } from "./use-firebase-auth";

export function useLogout() {
  const router = useRouter();
  const { getFirebaseAuth } = useFirebaseAuth();

  async function logout() {
    const auth = getFirebaseAuth();
    await signOut(auth);
    // Removes authenticated cookies
    await fetch("/api/logout", {
      method: "GET",
    });
    router.push("/");
  }

  return logout;
}
```
> P.S. As per the example repo, I don't actually any routes on this paths, which I assume is why I get the error

**Comment by awinogrodzki** [comment]
hey @MendyLanda! Thanks for reporting the issue.

What are the contents of your `/auth/logout` page?

**Comment by awinogrodzki** [comment]
Hello again @MendyLanda!

I hope you have already sorted the issue on your side.

I will close this one. If you have any further questions, please feel free to re-open :-)

**Comment by mason-stewart** [comment]
@awinogrodzki we're actually seeing this type of exception quite a bit both locally and in production, despite components being clearly marked as `"use client"` and doing similar things with session creation and deletion like in the OP's example. It looks like a typical Node.js exception where something is being server-side rendered and Node.js don't know how to parse a URL that's just a path. Confusingly, despite this error I'm still able to authenticate normally. I can't figure out what's causing these components to be SSR'd or what other error is occurring. Any clues about this?

**Comment by mason-stewart** [comment]
@awinogrodzki I was able to remove this error from the logs (again, everything was still working fine despite this error) by wrapping the function containing the fetch to `/api/login` in a `useEffect`. It seems as though Next 14.2 was somehow attempting SSR on my component despite `"use client"`. It's odd to me that in the `minimal` example in this repo, you clearly don't aren't wrapping your login page fetch in `useEffect`. Not sure if this is something worth documenting, but I figured it might help you or someone else who runs into this. Thanks for you work on this lib! 🎉
